.github/workflows/main.yml

```diff
@@ -17,6 +17,7 @@ jobs:
   windows-msbuild-debug:
     name: Run debug msbuild on Windows
     runs-on: windows-2019
+    timeout-minutes: 12
```

Comment:
Do you how long it normally takes to execute all these jobs? How much buffer do you have here?

Reply:
It's been 9-10 the last few I've looked at.  We can increase it if we want.

Reply:
Sorry, that comment (8-10) was just for the msbuild-debug task.  The others are similarly given a slight buffer based on their current run lengths.
My feeling was to try and keep them somewhat tight at first and open them up later if we need to.

Reply:
Please make sure that the buffer is large enough so we avoid frequent false positives.

Reply:
Sure.  Would you prefer we bump them up further now, or fix them as we get more experience?

Reply:
I'm fine either way as long as CI is stable. :) If you think that these are the right number, then go for it!

Reply:
I'm not a fan of flaky CI either, but I don't have enough data points to be certain on that yet I think.
I'm fine to be flexible about it and fix it up later if we need to.  If nothing else having tighter timeouts should force us to evaluate when something got introduced that is too long vs. something being flaky.